include/simdjson/generic/numberparsing.h

```diff
@@ -608,12 +608,15 @@ simdjson_unused simdjson_really_inline simdjson_result<uint64_t> parse_unsigned(
 
   // If there were no digits, or if the integer starts with 0 and has more than one digit, it's an error.
   int digit_count = int(p - start_digits);
-  if (digit_count == 0 || ('0' == *start_digits && digit_count > 1)) { return NUMBER_ERROR; }
-  if (!jsoncharutils::is_structural_or_whitespace(*p)) { return NUMBER_ERROR; }
+  if (digit_count == 0) { return INCORRECT_TYPE; }
+  if ('0' == *start_digits && digit_count > 1) { return NUMBER_ERROR; }
+  if (!jsoncharutils::is_structural_or_whitespace(*p)) {
+    return (*p == '.' || *p == 'e' || *p == 'E') ? INCORRECT_TYPE : NUMBER_ERROR;
+  }
 
   // The longest positive 64-bit number is 20 digits.
   // We do it this way so we don't trigger this branch unless we must.
-  if (digit_count > 20) { return NUMBER_ERROR; }
+  if (digit_count > 20) { return INCORRECT_TYPE; }
```

Comment:
Can this be merged with 
```
if (digit_count == 0) { return INCORRECT_TYPE; }
```
?
The following should be faster...

```
if ((digit_count <= 0) ||(digit_count > 20)) { return INCORRECT_TYPE; }
```
Because... you only have to check that `unsigned(digit_count-1)>=20`. So just one branch,